docs/protocol/network/messages/inventory.md

```diff
@@ -0,0 +1,163 @@
+
+## Inv message
+
+The `inv` message is used to advertise the knowledge of one or more objects (e.g. blocks, transactions, ...). The inventory message can be received unsolicited or in reply to a `get_blocks` message.
+
+The `inv` message consists of the message header with the command `INV` and as payload it contains a one more inventory entries:
+
+| Field     | Type       | Description                 |
+| --------- | :--------: | --------------------------- |
+| count     | u16        | number of inventory entries |
+| inventory | inv_vect[] | inventory vectors           |
+
+The type `inv_vect` (inventory vector) has the following data format:
+
+| Field | Type     | Description                                  |
+| ----- | :------: | -------------------------------------------- |
+| type  | u8       | type of object linked to the inventory entry |
+| count | [u32; 8] | hash of the object                           |
+
+The values of the type field are:
+
+| Value | Name  | Description                          |
+| ----- | :---: | ------------------------------------ |
+| `0`   | ERROR | Data with this number may be ignored |
+| `1`   | TX    | Hash is related to a transaction     |
+| `2`   | BLOCK | Hash is related to a block           |
+
+In the future, the type values may be extended in order to consider additional features.
+
+## Get data messages
+
+The messages `get_data` are used in order to request specific objects from other nodes. Usually, `get_data` messages are sent after receiving a `inv` (inventory) message and filtering the already known objects. The response to a `get_data` message can be a `block` or `tx` message.
+
+The `get_data` message consists of the message header with the command `GET_DATA` and its payload has the following format:
+
+| Field     | Type       | Description                 |
+| --------- | :--------: | --------------------------- |
+| count     | u16        | number of inventory entries |
+| inventory | inv_vect[] | inventory vectors           |
+
+## Block message
+
+The `block` message is used to transmit a single serialized block as response to a `get_data` message, which is requesting information given a block hash.
+
+The `block` message consists of the message header with the command `BLOCK` and the payload is formatted as follows:
+
+| Field     | Type         | Description                                                |
+| --------- | :----------: | ---------------------------------------------------------- |
+| header    | block_header | block header, as described in the [Block] section          |
+| txn_count | u32          | total number of transactions in the block                  |
+| txns      | tx[]         | block transactions as described in the transaction section |
+
+## Tx message
+
+Analogously, the `tx` message is used to transmit a single serialized transaction as response to a `get_data` message, which is requesting information given a transaction hash.
```

Comment:
Analogously, the `tx` message is used to transmit a single serialized transaction as a response to a `get_data` message, which is requesting information given a transaction hash.

Reply:
As above on line 142, I'm not sure I understand the use of "given" here.